src/librustc_interface/passes.rs

```diff
@@ -211,7 +171,7 @@ impl BoxedResolver {
             Err(resolver) => {
                 let resolver = &*resolver;
                 resolver.borrow_mut().access(|resolver| {
-                    ExpansionResult::from_resolver_ref(resolver)
+                    ExpansionResult::from_resolver_outputs(resolver.clone_outputs())
```

Comment:
This is why we can't have nice things.
This case puts the cloneability requirement on resolver outputs (https://github.com/rust-lang/rust/pull/65625/files#r336780545), but it actually never happens in `rustc`, only in `rustdoc` and perhaps other tools using `rustc_interface`.

Reply:
Why would cloning be needed? Is this that support for resolving things after `rustc_resolve` finishes?
In that case I'm not sure why we need to clone the *outputs* - is it to be able to keep the original resolver around?

Reply:
Not sure, `rustc_interface` is very hard to reason about. ¯\_(ツ)_/¯
It needs some investigation, but I'm not sure I'll get to it until the next weekend.

Reply:
This clone only happens with rustdoc since it needs the resolver available at a later stage. It would be nicer if we could extract the information rustdoc needs instead.

Reply:
Can we split the resolver up so that we can pass a `&CStore` later to the parts of it that we need to keep around? (i.e. the local module tree)

Reply:
We could make `expansion` query produce the final `CStore` structure and I said in my other comment. That won't help to get rid of the clone here though.

Reply:
I'd prefer if rustdoc didn't cause that clone to happen but instead did its resolving in some other way.

Reply:
It really feels like the new code is not _worse_, right? So we can definitely leave this for a future PR (filing an issue wouldn't hurt). I'd rather not block this PR on that cleanup.

Reply:
This clone was there before this PR.